### PR TITLE
update number of the corner of the second vertex

### PR DESCRIPTION
**Pull request by Davknapp**
The corner number was used wrongly in this example. We were using the vertex number of the vertices given to the tree instead of the vertices of the element. 

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

#### General
- [x] The reviewer executed the new code features at least once and checked the results manually

- [x] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [x] New source/header files are properly added to the Makefiles
- [x] The code is well documented
- [x] All function declarations, structs/classes and their members have a proper doxygen documentation
- [x] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)

#### Tests
- [x] The code is covered in an existing or new test case using Google Test

#### Github action

- [x] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)
- [ ] All tests pass (in various configurations, this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

#### Scripts and Wiki

- [ ] If a new directory with source-files is added, it must be covered by the `script/find_all_source_files.scp` to check the indentation of these files.
- [ ] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

#### Licence

- [x] The author added a BSD statement to `doc/` (or already has one)

### CODE REVIEW

**Review by lukasdreyer — CHANGES_REQUESTED** [review]
I'd love to see a test that would have caught this

**Comment by Davknapp** [comment]
> I'd love to see a test that would have caught this

I've extended the point inside check, such that it would've failed.
During debugging I also found another bug, which lead to a failed vtk-output for axis aligned geometries. To catch this bug I opend up the issue #980 , because I think that it requires more detailed testing which would be out of the scope of this PR.

File: test/t8_geometry/t8_geometry_implementations/t8_gtest_geometry_linear.cxx [hunk 1]
```diff
@@ -25,88 +25,142 @@
...
+  void
+  SetUp () override
+  {
+    const int geom_int = std::get<0> (GetParam ());
+    eclass = std::get<1> (GetParam ());
+    t8_cmesh_init (&cmesh);
+    if (geom_int == T8_GEOMETRY_TYPE_LINEAR_AXIS_ALIGNED
+        && !(eclass == T8_ECLASS_LINE || eclass == T8_ECLASS_QUAD || eclass == T8_ECLASS_HEX)) {
+      GTEST_SKIP ();
+    }
```
Comment (lukasdreyer) [inline test/t8_geometry/t8_geometry_implementations/t8_gtest_geometry_linear.cxx:57]: I would rather just initialize the testsuite only with the line, quad and hex eclass

File: test/t8_geometry/t8_geometry_implementations/t8_gtest_geometry_linear.cxx [hunk 2]
```diff
@@ -25,88 +25,142 @@
...
+      vertices[3] = vertices[3 * (num_vertices - 1)];
+      vertices[4] = vertices[3 * (num_vertices - 1) + 1];
+      vertices[5] = vertices[3 * (num_vertices - 1) + 2];
+
+      t8_cmesh_register_geometry<t8_geometry_linear_axis_aligned> (cmesh, t8_eclass_to_dimension[eclass]);
+      break;
+    default:
+      break;
+    }
+    t8_cmesh_set_tree_vertices (cmesh, 0, vertices, t8_eclass_num_vertices[eclass]);
```
Comment (lukasdreyer) [inline test/t8_geometry/t8_geometry_implementations/t8_gtest_geometry_linear.cxx:84]: For the axis aligned geometry, only two vertices should be set

File: test/t8_geometry/t8_gtest_point_inside.cxx [hunk 3]
```diff
@@ -301,6 +303,12 @@ TEST_P (geometry_point_inside, test_point_inside)
 
         num_in += point_is_inside ? 1 : 0;
       }
+      for (int side = 0; side < min_points_outside; ++side) {
+        test_point[3 * (num_points + side)] = (side == 0) ? -2.0 : (side == 1) ? 2.0 : -0.1;
+        test_point[3 * (num_points + side) + 1] = (side == 2) ? -2.0 : (side == 3) ? 2.0 : 0.3;
+        test_point[3 * (num_points + side) + 2] = (side == 4) ? -2.0 : (side == 5) ? 2.0 : 0.15;
+        point_is_inside[num_points + side] = false;
+      }
```
Comment (lukasdreyer) [inline test/t8_geometry/t8_gtest_point_inside.cxx:311]: What is happening here? Please document

**Review by lukasdreyer — CHANGES_REQUESTED** [review]

File: test/t8_geometry/t8_geometry_implementations/t8_gtest_geometry_linear.cxx (hunk 1 above)
Comment (Davknapp) [reply test/t8_geometry/t8_geometry_implementations/t8_gtest_geometry_linear.cxx:57]: But for the linear geometry we want to test all eclasses.

**Review by lukasdreyer — APPROVED** [review]

**Pull request merged**